e2e/framework/cluster.go

```diff
@@ -30,6 +30,18 @@ func CreateCoreCluster(cl client.Client, clusterName, infraClusterKind string) *
 		},
 	}
 
+	if infraClusterKind == "VSphereCluster" {
+		host, port, err := GetControlPlaneHostAndPort(cl)
+		if err != nil {
+			Expect(err).ToNot(HaveOccurred())
+		}
+
+		cluster.Spec.ControlPlaneEndpoint = clusterv1.APIEndpoint{
+			Host: host,
+			Port: port,
+		}
+	}
+
```

Comment:
Do we need this only for VSphere?
Ideally we should avoid special cases in the framework.

Reply:
Yes, AFAIK, this is necessary for vSphere only. I haven't tried what it would do with other providers if we make this setting global.
VSphere needs it so that the management cluster correctly communicates with vCenter to create resources, edit, etc.

Reply:
It looks like we are setting this up also for other platforms (i.e. [AWS](https://github.com/openshift/cluster-capi-operator/pull/126/files#diff-872c7f24229a19d1d00c03edf29459f9fd8c1edcb92fcaa61644ad8a1370e22cR116-R119)), but we are doing this in the `<provider>_test.go` rather than in the framework.

I think it'd be nice to try and centralise this for all the providers in the framework (by removing the `if infraClusterKind == x`, and removing that set up in the other `<provider>_test.go`). If that doesn't work, I'd move the VSPhere one in the `vsphere_test.go` similarly to how it is already done in the AWS one.

Reply:
But I don't think we are setting it up for the core cluster object for any other provider.

Reply:
Ah it's on the core cluster also. That's quite curious.
And vsphere E2Es don't work if this is not set on the core `Cluster` object?